### PR TITLE
PTX-23991  Deleting Pools after modifying STC with disabling CSI topology 

### PR DESCRIPTION
**Pull request by sayalasomayajula-px**
…logy

**What this PR does / why we need it**:
This PR will disable CSI topology and will delete a pool and check if it is deleted successfully, and once that is done will add the pool again to the node and check if it is added successfully and also toggles back the csi topology to true.
**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/Users%2Fkrishna%2Ftp-pxe-byoc/detail/tp-pxe-byoc/3344/pipeline/
https://jenkins.pwx.dev.purestorage.com/blue/organizations/jenkins/Users%2Fkrishna%2Ftp-pxe-byoc/detail/tp-pxe-byoc/3339/pipeline

### CODE REVIEW

File: tests/basic/pure_test.go [hunk 1]
```diff
@@ -5000,4 +5002,115 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
...
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			stc.Spec.CSI.Topology.Enabled = false
+			pxOperator := operator.Instance()
+			_, err = pxOperator.UpdateStorageCluster(stc)
+			log.FailOnError(err, "Failed to update the storage cluster")
+			log.InfoD("Validating csi pods are deleted")
+
+			_, err = task.DoRetryWithTimeout(checkPodIsDeleted, 15*time.Minute, 30*time.Second)
+			Expect(err).NotTo(HaveOccurred(), "Failed to rescan specs from %s", Inst().SpecDir)
```
Comment (lsrinivas-pure) [inline tests/basic/pure_test.go:5076]: same here

File: tests/basic/pure_test.go [hunk 2]
```diff
@@ -5000,4 +5002,115 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
...
+			log.InfoD("Update STC, is csi topology enabled Now?: %t", stc.Spec.CSI.Topology.Enabled)
+		})
+		stepLog = "Delete the pool on a particular node and validate if it is Deleted Successfully"
+		Step(stepLog, func() {
+			log.InfoD(stepLog)
+			nodeForPoolDelete = append(nodeForPoolDelete, stNodes[rand.Intn(len(stNodes))])
+			log.InfoD("Deleting the pool on the node [%v]", nodeForPoolDelete[0].Name)
+			deletePoolAndValidate(nodeSelected, fmt.Sprintf("%d", nodePool.ID))
+			log.FailOnError(err, "Failed to delete the pool on the node [%v]", nodeForPoolDelete[0].Name)
```
Comment (lsrinivas-pure) [inline tests/basic/pure_test.go:5085]: where is err coming form?

File: tests/basic/pure_test.go [hunk 3]
```diff
@@ -5000,4 +5002,115 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
...
+			log.InfoD(stepLog)
+			stc.Spec.CSI.Topology = &v12.CSITopologySpec{
+				Enabled: true,
+			}
+			pxOperator := operator.Instance()
+			_, err = pxOperator.UpdateStorageCluster(stc)
+			log.FailOnError(err, "Failed to update the storage cluster")
+			log.InfoD("Validating csi pods are deployed")
+			_, err = task.DoRetryWithTimeout(checkPodIsDeleted, 15*time.Minute, 30*time.Second)
+			Expect(err).NotTo(HaveOccurred(), "Failed to rescan specs from %s", Inst().SpecDir)
```
Comment (lsrinivas-pure) [inline tests/basic/pure_test.go:5106]: use dash verifiaction or log.FailonError

File: tests/basic/pure_test.go [hunk 4]
```diff
@@ -5000,4 +5002,115 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
...
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("disablecsitopologyandpooldelete-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+		checkPodIsDeleted := func() (interface{}, bool, error) {
+			csiLabels := make(map[string]string)
+			csiLabels["app"] = "px-csi-driver"
+			pods, err := k8sCore.GetPods(volDriverNamespace, csiLabels)
+			Expect(err).NotTo(HaveOccurred())
```
[the remark this answers is not in the document]
Comment (sayalasomayajula-px) [reply tests/basic/pure_test.go]: changed

File: tests/basic/pure_test.go (hunk 1 above)
Comment (sayalasomayajula-px) [reply tests/basic/pure_test.go]: changed

File: tests/basic/pure_test.go (hunk 2 above)
Comment (sayalasomayajula-px) [reply tests/basic/pure_test.go]: removed it

File: tests/basic/pure_test.go [hunk 5]
```diff
@@ -5000,4 +5002,114 @@ var _ = Describe("{RebootingNodesWhileFADAvolumeCreationInProgressUsingZones}",
...
+		}
+		for i := 0; i < Inst().GlobalScaleFactor; i++ {
+			contexts = append(contexts, ScheduleApplications(fmt.Sprintf("disablecsitopologyandpooldelete-%d", i))...)
+		}
+		ValidateApplications(contexts)
+		defer appsValidateAndDestroy(contexts)
+		checkPodIsDeleted := func() (interface{}, bool, error) {
+			csiLabels := make(map[string]string)
+			csiLabels["app"] = "px-csi-driver"
+			pods, err := k8sCore.GetPods(volDriverNamespace, csiLabels)
```
Comment (lsrinivas-pure) [inline tests/basic/pure_test.go:5058]: return the error here and handle it in test case

**Review by lsrinivas-pure — APPROVED** [review]
LGTM

**Pull request merged**